python/cudf_polars/cudf_polars/dsl/expr.py

```diff
@@ -826,6 +827,46 @@ def do_evaluate(
                     else prefix.obj,
                 )
             )
+        elif self.name == pl_expr.StringFunction.Strptime:
+            # TODO: ignores ambiguous
+            format, strict, exact, cache = self.options
+            col = self.children[0].evaluate(df, context=context, mapping=mapping)
+
+            if format is None:
+                raise NotImplementedError("Strptime requires a format string")
+
+            if cache:
+                raise NotImplementedError("Strptime cache is a CPU feature")
+
+            if not exact:
+                raise NotImplementedError("Strptime does not support exact=False")
+            not_timestamps = plc.unary.unary_operation(
+                plc.strings.convert.convert_datetime.is_timestamp(
+                    col.obj, format.encode()
+                ),
+                plc.unary.UnaryOperator.NOT,
+            )
```

Comment:
We can avoid the negation, because what you want is that `all(is_timestamp(col))`.

So:

```
is_timestamp = is_timestamp(...)

if strict:
    all_timestamp = plc.reduce.reduce(is_timestamp, plc.aggregation.all(), bool)
    if not plc.interop.to_arrow(all_timestamp).as_py():
        raise ...
else:
    mask = plc.unary.unary_operation(is_timestamp, NOT)
    bitmask, null_count = transform.bools_to_mask(mask) # Need to expose this
    result = col.obj.with_mask(bitmask, null_count)
    to_timestamps(...)
```

Reply:
Is this something we want to do?

Reply:
I think it's worth updating to use a `reduce` so that we can avoid the negation in the `strict` case. I am a bit hesitant to implement `bools_to_mask` as part of this PR merely for scope reasons. I think it would benefit us to take our time with a comprehensive set of tests for `bools_to_mask` especially considering the way libcudf pads mask buffers in edge cases and so forth.